keras/backend/tensorflow_backend.py

```diff
@@ -575,10 +575,20 @@ def maxpool2d(x, pool_size, strides=(1, 1),
         # TH kernel shape: (depth, input_depth, rows, cols)
         # TF kernel shape: (rows, cols, input_depth, depth)
         x = tf.transpose(x, (0, 2, 3, 1))
-        x = tf.nn.max_pool(x, pool_size, strides, padding=padding)
+        if pool_mode=='max':
```

Comment:
This should be refactored to eliminate code redundancy. The conditional flow should affect the reshape ops.